### PR TITLE
Fixes #808

### LINKED ISSUE #808: Broken methods(Base.show)
```julia
using Zygote
methods(Base.show)
```
is broken, causes
```julia
[97] Error showing value of type Base.MethodList:
ERROR: UndefVarError: S not defined
Stacktrace:
 [1] show(::IOContext{Base.GenericIOBuffer{Array{UInt8,1}}}, ::Type{
SYSTEM (REPL): showing an error caused an error
ERROR: UndefVarError: S not defined
Stacktrace:
 [1] show(::IOContext{REPL.Terminals.TTYTerminal}, ::Type{
SYSTEM (REPL): caught exception of type UndefVarError while trying to handle a nested exception; giving up
```

### PR DESCRIPTION
**Pull request by racinmat**
Fixes #808 
hopefully it's done right.
Could this PR be labeled by `hacktoberfest-accepted`? 
Thanks.

### CODE REVIEW

File: src/compiler/show.jl [hunk 1]
```diff
@@ -10,4 +10,4 @@ end
 
 Base.show(io::IO, j::Pullback{S}) where S = print(io, "∂($(funcname(S.parameters[1])))")
 
-Base.show(io::IO, P::Type{<:Pullback{S}}) where S<:Tuple = print(io, "typeof(∂($(funcname(S.parameters[1]))))")
+Base.show(io::IO, P::Type{<:Pullback{S}}) where S<:Tuple = print(io, "typeof(∂($(funcname((@isdefined S) ? S.parameters[1] : nothing))))")
```
Comment (DhairyaLGandhi) [inline src/compiler/show.jl:13]:
```suggestion
Base.show(io::IO, P::Type{<:Pullback{S}}) where S<:Tuple = print(io, "typeof(∂($(funcname(@isdefined(S) ? S.parameters[1] : nothing))))")
```

**Review by DhairyaLGandhi — COMMENTED** [review]
We should also test this

**Review by racinmat — COMMENTED** [review]
Any idea for good test? Because I'm afraid this happens only when listing methods, not during calling the function.
Should I just test that `methods(Base.show)` does not crash or is there soemthin better we can do?

**Comment by racinmat** [comment]
Testcase added, I couldn't figure out better way than this, I'm looking for ideas how to test it better.
Btw, could you please change the label to `hacktoberfest-accepted`.
My bad, I misspelled it in my PR.

**Comment by DhairyaLGandhi** [comment]
Yeah perhaps not having it error would be enough. Thanks!

Could you add it in a test set please? Seems like a pretty standalone fix, so differentiating it in the tests

**Comment by racinmat** [comment]
Ok, I changed tests, now it just tests that printed buffer is not empty

**Comment by racinmat** [comment]
@DhairyaLGandhi

**Comment by racinmat** [comment]
@CarloLucibello  is this mergeable or something more to do?

**Comment by CarloLucibello** [comment]
thanks!

bors r+

**Pull request merged**